src/identity/identity-wallet.ts

```diff
@@ -409,6 +423,26 @@ export class IdentityWallet implements IIdentityWallet {
 
     credential.proof = [mtpProof];
 
+    if (opts.revocationOpts.type === CredentialStatusType.Iden3ReverseSparseMerkleTreeProof) {
+      const revocationTree = await this._storage.mt.getMerkleTreeByIdentifierAndType(
+        did.string(),
+        MerkleTreeType.Revocations
+      );
+
+      const rootOfRootsTree = await this._storage.mt.getMerkleTreeByIdentifierAndType(
+        did.string(),
+        MerkleTreeType.Roots
+      );
+
+      const trees: TreesModel = {
+        state: currentState,
+        claimsTree: claimsTree,
+        revocationTree: revocationTree,
+        rootsTree: rootOfRootsTree
+      };
+      await pushHashesToRHS(currentState, trees, opts.revocationOpts.id);
```

Comment:
can't we just call publishStateToRHS here?

Reply:
Yes. I did it